src/components/dialog/dialog-container.ts

```diff
@@ -0,0 +1,70 @@
+import {Component, ComponentRef, ViewChild, AfterViewInit} from '@angular/core';
+import {BasePortalHost, ComponentPortal, TemplatePortal} from '../../core/portal/portal';
```

Comment:
Even further; it should be `@angular2-material/core` since it will be bundled like Angular.